### PR TITLE
Skip experiential-memory 0*inf utility decay

### PR DESCRIPTION
**Pull request by Kayvan-Zahiri**
## Summary
- Experiential memory decays eviction utilities each step with a scale that may be 0. IEEE 0*inf is NaN and the finite-state guard freezes writes.
- Skip that product before aging slots. When decay is 0, treat already-infinite utilities as discarded in finite-state and query eligibility checks.

## Test plan
- [x] `test_zero_utility_decay_does_not_multiply_inf_utilities`
- [x] `tests/test_experiential_memory.py` (54 passed)
- [x] ruff on the touched files

Made with Cursor. No Slop measured-run receipt (not an approved Codex or Claude Code client).

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by kenjohnscreates — APPROVED** [review]
## Sanitized alias stays where it should; one weakening worth documenting

Approving. The part of this change that usually goes wrong is handled correctly, and I want to record the check so it isn't repeated.

**The sanitized copy never enters arithmetic.** When a PR introduces a checked alias, the failure mode is that the *unsanitized* name keeps being used somewhere downstream. Here `utilities_for_check` appears only in the validity conjunction of `_state_is_valid` and in the `sane_rows` / finiteness conjunction of `_query_jit` — the raw `entries.utilities` never reaches query scoring.

There is one place the raw values do enter arithmetic, in the eviction score:

```python
eviction_utilities = jnp.where(current_entries.utility_available, current_entries.utilities, 0.0)
retention_score = utility_weight / weight_scale * eviction_utilities + ...
```

That is safe, but only because of ordering — `write()` dispatches to `self._write_advanced(self._advance(state), entry)`, and `_advance` has already replaced the utilities with exact zeros under `utility_decay == 0.0`. So eviction sees finite values. It's worth a comment at the eviction site, since the safety depends on a call-ordering invariant a reader can't see locally.

**The weakening, stated plainly.** `_utilities_for_finite_guard` returns a zero-filled copy whenever `utility_decay == 0.0`, which is an allowed config (`_validated_config_float("utility_decay", ..., lower=0, upper=1)`). So for those configs `_state_is_valid` no longer detects non-finite utilities, and a state corrupted by something other than the decay path is accepted rather than triggering abstention. `jnp.isfinite` is false for NaN as well as inf, so NaN corruption is masked the same way.

The blast radius is bounded — the masked values are zeroed by `_advance` before any arithmetic, and every other payload field is still checked — so this is a deliberate narrowing of a corruption detector rather than a hole. But it is a real reduction in what the dynamic invariant covers, and the docstring ("Treat poisoned eviction utility as discarded when decay zeroes it") reads as a description of intent rather than a warning that detection is being dropped. Worth saying so explicitly there.

**Comment by lalalune** [comment]
Closing after implementation and regression review. The test injects infinite utility values, and the patch weakens both write validity and read/query row validity based on decay. Query does not first decay these utilities; configuration-dependent acceptance of corrupted stored metadata is not a sound recovery contract.
